src/journal/views.py

```diff
@@ -359,6 +359,50 @@ def print_article(request, identifier_type, identifier):
     return render(request, template, context)
 
 
+@has_journal
+@keyword_page_enabled
+def keywords(request):
+    """
+    Renders a list of keywords
+    :param request: HttpRequest object
+    :return: a rendered template
+    """
+    keywords = request.journal.article_keywords()
+
+    template = 'journal/keywords.html'
+    context = {
+        'keywords': keywords,
+    }
+
+    return render(request, template, context)
+
+
+@has_journal
+@keyword_page_enabled
+def keyword(request, keyword_id):
+    """
+    Displays a list of articles that use a given keyword.
+    :param request: HttpRequest object
+    :param keyword_id: Keyword object PK
+    :return: a rendered template
+    """
+    keyword = get_object_or_404(submission_models.Keyword, pk=keyword_id)
+    articles = submission_models.Article.objects.filter(
```

Comment:
Now it can be `request.journal.published_articles().filter(keywords__pk=keyword.pk)` to avoid repeating the published articles logic :)